server/controllers/finance/journal/index.js

```diff
@@ -276,7 +280,28 @@ function editTransaction(req, res, next) {
         // row = transformColumns(row);
         transaction.addQuery(INSERT_JOURNAL_ROW, [row]);
       });
+      return selectTransDate(rowsChanged);
+    })
+    .then((trans_date) => {
+      transDate = trans_date;
+      return FiscalService.lookupFiscalYearByDate(transDate);
+    })
+    .then((result) => {
+      if (transDate) {
+        if (!result.length) {
+          throw new BadRequest('Invalid Fiscal Year', 'POSTING_JOURNAL.ERRORS.NO_FISCAL_FOR_DATE');
+        }
+
+        if (result[0].locked) {
+          throw new BadRequest('Closed Fiscal Year', 'POSTING_JOURNAL.ERRORS.CLOSED_FISCAL_YEAR');
+        }
 
+        _.each(rowsChanged, (row) => {
```

Comment:
This function alters data in the transaction.   @sfount attempted to avoid [spaghetti code](https://en.wikipedia.org/wiki/Spaghetti_code) by putting all logic that changes the transaction in the function `transformColumns()`.   This code belongs there.